### ISSUE TITLE
make Core a QML module

### ISSUE THREAD

**Issue by maidamai0** [qCring/QuickVtk#49]
I noticed all QML types like **vtk.actor** are registered to qmlengine by *qmlRegisterType* family functions in c++ code.Have you cansidered to make them a standalone QML module use **qmldir** so everyone can use it in QML like the native QML types without register it in c++ code.?

**Comment by qCring** [comment]
I'm not sure I understand correctly. The C++ classes are always necessary in order to use VTK types in QML. So you will always have a dependency to the C++ implementation and therefore the QuickVtk Core library. This is why I deliberately left type registration on the C++ side. 

I assume you mean [QML Plugins](https://doc.qt.io/qt-5/qtqml-modules-cppplugins.html), right? That would be possible but the only advantage I can see is a better integration with QtCreator and `.pro` files. I'm not saying that this is not relevant. But I think it makes more sense to stick to [CMake](https://cmake.org/) to handle project dependencies. Plus, [QML Plugins](https://doc.qt.io/qt-5/qtqml-modules-cppplugins.html) are also just libraries, so I'm not too sure if moving the QuickVtk Core to such a plugin would really change things. But please correct me if I'm wrong since I didn't really spent too much time on this topic. I'm always happy for every single bit of feedback and fresh ideas!

**Comment by maidamai0** [comment]
Yes, I mean [QML Plugins](https://doc.qt.io/qt-5/qtqml-modules-cppplugins.html).If **QuickVtk Core** is made to a [QML Plugins](https://doc.qt.io/qt-5/qtqml-modules-cppplugins.html), it will act like a native QML type just like [Rectangle QML Type](https://doc.qt.io/qt-5/qml-qtquick-rectangle.html).Everyone who wants to use vtk in QML just need copy the [qmldir](https://doc.qt.io/qt-5/qtqml-modules-qmldir.html) file and the library file and add a path to [QML Import Path](https://doc.qt.io/qt-5/qtqml-syntax-imports.html#qml-import-path).

Just like creating a QML bindings for vtk and I find this is convinent.With [qmlplugindump](https://doc.qt.io/qtcreator/creator-qml-modules-with-plugins.html#generating-qmltypes-files) you can give QtCreator code completion, again, like the native QML types :blush:.

**Comment by qCring** [comment]
Code completion is a good point though. Also, I just searched for Qt package managers (something like [NuGet](https://www.nuget.org/) but for Qt) and found the [qpm](http://www.qpm.io/) project ([GitHub link](https://github.com/Cutehacks/qpm)). So yeah, there are some good reasons to consider QML Plugins. I'll keep this issue open for now and come back to this later. Right now, I'm working on a somewhat big refactoring to clean up the project and the GitHub Pages project documentation. Might take a bit but things are looking good. I'm planning on this to be done in the next week(s). After that I'll have some time for further research!
